src/routes/transactions/pipes/delete-transaction.validation.pipe.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable, PipeTransform } from '@nestjs/common';
+import { ValidateFunction } from 'ajv';
+import { GenericValidator } from '@/validation/providers/generic.validator';
+import { JsonSchemaService } from '@/validation/providers/json-schema.service';
+import { DeleteTransactionDto } from '@/routes/transactions/entities/delete-transaction.dto.entity';
+import {
+  DELETE_TRANSACTION_DTO_SCHEMA_ID,
+  deleteTransactionDtoSchema,
+} from '@/routes/transactions/entities/schemas/delete-transaction.dto.schema';
+
+@Injectable()
+export class DeleteTransactionDtoValidationPipe
+  implements PipeTransform<unknown, DeleteTransactionDto>
+{
+  private readonly isValid: ValidateFunction<DeleteTransactionDto>;
+
+  constructor(
+    private readonly genericValidator: GenericValidator,
+    private readonly jsonSchemaService: JsonSchemaService,
+  ) {
+    this.isValid = this.jsonSchemaService.getSchema(
+      DELETE_TRANSACTION_DTO_SCHEMA_ID,
+      deleteTransactionDtoSchema,
+    );
+  }
+  transform(data: unknown): DeleteTransactionDto {
+    return this.genericValidator.validate(this.isValid, data);
```

Comment:
I think we should set the error status (400 - bad request) here, what do you think?

It's done that way in other ValidationPipes and I think it makes sense to communicate that a validation error at this level is a client-side error (400).

Reply:
Agreed. I've added it in 731824e9750feec78d152c4caa1b4c0d592b326b but in a type safe way (adjusted in other PRs).